wemake_python_styleguide/visitors/ast/statements.py

```diff
@@ -498,3 +508,108 @@ def _check_tuple_arguments_types(
             if isinstance(arg, self._no_tuples_collections):
                 self.add_violation(NotATupleArgumentViolation(node))
                 break
+
+
+@final
+class UnsafeGeneratorExpressionVisitor(BaseNodeVisitor):
+    """Checks if a generator expression is safe."""
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Creates the lists that would hold the variables."""
+        super().__init__(*args, **kwargs)
+        self.inside_variables: List[str] = []
+        self.created_variables: List[str] = []
+        self.affecting_variables: List[str] = []
+
+    def visit(self, node: ast.AST) -> None:
```

Comment:
Why don't you use `visit_any_assign` here? It does the same thing.

Reply:
I didn't consider it 🤔, I'm changing it!